### PR TITLE
Improve lint setup

### PR DESCRIPTION
**Pull request by mehulkar**
Before:
```
digraph {
	compound = "true"
	newrank = "true"
	subgraph "root" {
		"[root] //#lint" -> "[root] ___ROOT___"
	}
}
```

After:
```
digraph {
	compound = "true"
	newrank = "true"
	subgraph "root" {
		"[root] @turbo/codemod#lint" -> "[root] ___ROOT___"
		"[root] @turbo/pack-test-harness#lint" -> "[root] ___ROOT___"
		"[root] @vercel/experimental-nft#lint" -> "[root] ___ROOT___"
		"[root] @vercel/experimental-nft-next-plugin#lint" -> "[root] ___ROOT___"
		"[root] @vercel/turbo-tracing-test-app#lint" -> "[root] ___ROOT___"
		"[root] @vercel/turbopack-next#lint" -> "[root] ___ROOT___"
		"[root] @vercel/turbopack-node#lint" -> "[root] ___ROOT___"
		"[root] @vercel/turbopack-runtime#lint" -> "[root] ___ROOT___"
		"[root] @vercel/webpack-nft#lint" -> "[root] ___ROOT___"
		"[root] benchmark#lint" -> "[root] ___ROOT___"
		"[root] cargo-sweep-action#lint" -> "[root] ___ROOT___"
		"[root] cli#lint" -> "[root] ___ROOT___"
		"[root] create-turbo#lint" -> "[root] ___ROOT___"
		"[root] docs#lint" -> "[root] ___ROOT___"
		"[root] eslint-config-turbo#lint" -> "[root] ___ROOT___"
		"[root] eslint-plugin-turbo#lint" -> "[root] ___ROOT___"
		"[root] next-dev-tests#lint" -> "[root] ___ROOT___"
		"[root] tsconfig#lint" -> "[root] ___ROOT___"
		"[root] turbo-ignore#lint" -> "[root] ___ROOT___"
		"[root] turbo-test-utils#lint" -> "[root] ___ROOT___"
		"[root] turbo-types#lint" -> "[root] ___ROOT___"
		"[root] turbo-utils#lint" -> "[root] ___ROOT___"
	}
}
```

### CODE REVIEW

File: turbo.json [hunk 1]
```diff
@@ -5,30 +5,29 @@
       "outputs": ["coverage/**/*"],
       "dependsOn": ["^build"]
     },
-    "//#lint": {},
+
+    // lint tasks
+    "lint": {},
```
Comment (nathanhammond) [inline turbo.json:10]: We're not using `typescript-eslint` so this works! 👍

(Linting is constrained to single files.)

**Review by nathanhammond — APPROVED** [review]

File: turbo.json (hunk 1 above)
Comment (mehulkar) [reply turbo.json:10]: Just to make sure I understand, are you saying we'd need a `dependsOn` if we were using `typescript-eslint`? Or are you referring to the fact that `lint`, rather then `//#lint` is kicking off the lint tasks now, and somehow `typescript-eslint` is relevant to that?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -1049,7 +1049,10 @@ jobs:
           github-token: "${{ secrets.GITHUB_TOKEN }}"
 
       - name: Format check
-        run: pnpm -- turbo run check:fmt
+        # This is the syntax for running the `//#lint` task specifically in turbo.json
+        # It runs the taplo check on taplo.toml, and prettier check across the whole repo.
+        # TODO: run prettier in individual workspaces instead of globally.
```
Comment (mehulkar) [inline .github/workflows/test.yml:1054]: fixed in https://github.com/vercel/turbo/pull/3925

File: turbo.json [hunk 3]
```diff
@@ -5,30 +5,29 @@
...
+    // root lint task runs toml check and prettier
+    // TODO: run prettier in individual workspaces instead of globally.
```
Comment (mehulkar) [inline turbo.json:12]: fixed in https://github.com/vercel/turbo/pull/3925

File: turbo.json (hunk 1 above)
Comment (nathanhammond) [reply turbo.json:10]: Yeah, just a `dependsOn`. If we were using `typescript-eslint` we would have to get builds to run first.

**Review by nathanhammond — APPROVED** [review]

**Comment by mehulkar** [comment]
2 failing tests, but I think they are not relevant here.

CleanShot 2023-02-23 at 09 21 17@2x

**Comment by mehulkar** [comment]
Confirmed these are not related with @alexkirsz

**Pull request merged**